src/client/editor/nodes/DirectionalLightNode.js

```diff
@@ -0,0 +1,90 @@
+import * as THREE from "three";
+import EditorNodeMixin from "./EditorNodeMixin";
+import Picker from "../objects/Picker";
+import PhysicalDirectionalLight from "../objects/PhysicalDirectionalLight";
+import SpokeDirectionalLightHelper from "../helpers/SpokeDirectionalLightHelper";
+import serializeColor from "../utils/serializeColor";
+
+export default class DirectionalLightNode extends EditorNodeMixin(PhysicalDirectionalLight) {
+  static legacyComponentName = "directional-light";
+
+  static nodeName = "Directional Light";
+
+  static async deserialize(editor, json) {
+    const node = await super.deserialize(editor, json);
+
+    const { color, intensity, castShadow } = json.components.find(c => c.name === "directional-light").props;
+
+    node.color.set(color);
+    node.intensity = intensity;
+    node.castShadow = castShadow;
+
+    return node;
+  }
+
+  constructor() {
+    super();
+
+    this.picker = new Picker();
+    this.add(this.picker);
+
+    this.helper = new SpokeDirectionalLightHelper(this);
+    this.add(this.helper);
+  }
+
+  onChange() {
+    this.helper.update();
+  }
+
+  copy(source, recursive) {
+    super.copy(source, false);
+
+    this.helper.update();
+
+    if (recursive) {
+      for (const child of source.children) {
+        if (child !== this.helper && child !== this.picker) {
+          const clonedChild = child.clone();
+          this.add(clonedChild);
+        }
+      }
+    }
+
+    return this;
+  }
+
+  serialize() {
+    const json = super.serialize();
```

Comment:
these `serialize` and `prepareForExport` routines don't seem completely DRY. doesn't matter that much but might be worth extracting functions